src/Unit.hs

```diff
@@ -89,7 +110,7 @@ infix 4 .>=.
 -- ** Formatting
 
 formatWith :: Unit dim u => (Maybe Int -> u a -> ShowS) -> Maybe Int -> u a -> String
-formatWith with n u = with n u (getConst (suffix `asTypeOf` (u <$ Const ('x':))) "")
+formatWith with n u = with n u (showChar ' ' (getConst (suffix `asTypeOf` (u <$ Const ('x':))) ""))
```

Comment:
Dimensionless quantities have unit 1, so we add a space to disambiguate, just in case.